doc: config_and_build: add table for CMake options (#14518)
nrfconnect/sdk-nrf: doc/nrf/config_and_build/configuring_app/kconfig/index.rst

      See :ref:`zephyr:menuconfig` in the Zephyr documentation for instructions on how to run menuconfig or guiconfig.
      To locate a specific configuration option, use the **Jump to** field.

.. _configuration_temporary_change_single_build:

Temporary Kconfig changes for a single build
============================================

You can also apply temporary Kconfig changes to a single build by :ref:`providing the value of a chosen Kconfig option as a CMake option <cmake_options>`.

> The Kconfig option configuration will be available until you clean the build directory pristinely.

Reviewer: ```suggestion
The Kconfig option configuration will be available until you clean the build directory.
```
The cleaning itself doesn't have to be pristine.

Author: Yes, but the option is being applied until you clean pristinely.

Usage of this word here was very adamantly defended by Torsten Rasmussen when I was working on this piece of docs around 2 years ago, so I will keep this pristinely.

.. _configuration_permanent_change:

Permanent Kconfig changes
*************************

To configure your application and maintain the configuration after you clean the build directory pristinely, you need to specify the configuration in one of the permanent configuration files.
In most of the cases, this means editing the default :file:`prj.conf` file of the application, but you can also use an extra Kconfig fragment file.

> The Kconfig fragment files are configuration files used for building an application image with or without software support from specific Kconfig options.

Reviewer: ```suggestion
The Kconfig fragment files are configuration files used for building an application image with or without software support that is enabled by specific Kconfig options.
```

In these files, you can specify different values for configuration options that are defined by a library or board, and you can add configuration options that are specific to your application.

> Examples include things like whether to add networking support or which drivers are needed by the application.

Reviewer: ```suggestion
Examples include whether to add networking support, or which drivers are needed by the application.
```

> Kconfig fragments are applied on top of the default :file:`prj.conf` file, use the `.conf` file extension, and can often have an ``overlay-`` prefix in their name.

Reviewer: ```suggestion
Kconfig fragments are applied on top of the default :file:`prj.conf` file, use the :file:`.conf` file extension, and can often have an ``overlay-`` prefix in their name.
```

> When they are board-specific, they are are placed in the :file:`boards` folder, are named :file:`<BOARD>.conf`, and they are applied on top of the default Kconfig file for the chosen board.

Reviewer: ```suggestion
When they are board-specific, they are placed in the :file:`boards` folder, are named :file:`<board>.conf`, and are applied on top of the default Kconfig file for the specified board.
```
